docs/setting-up-local-development.md

```diff
@@ -2,73 +2,187 @@
+
+1. Run exercism.io on your computer directly by [installing the required software](#running-exercism-io-directly-on-your-computer).
+2. Run exercism.io [in a Docker container via Vagrant](#running-exercism-io-in-a-docker-container-via-vagrant).
+
+### Running exercism.io directly on your computer
+
+These instructions assume you're using a package manager for your OS:
+
+- Windows — [Chocolatey](https://chocolatey.org/install)
+- Mac OS X — [Homebrew](http://brew.sh)
+- Linux — (whichever comes with your distro, we'll refer to `apt-get`)
+
+#### Installing Ruby
+
+(see [`Gemfile`](https://github.com/exercism/exercism.io/blob/master/Gemfile#L3) for exact version of Ruby to install).
+
+- Windows
+
+  ```
+  C:> choco install ruby --version X.Y.Z
+  ```
+- Mac OS X
+
+  ```
+  $ brew update
+  $ brew install rbenv
+  $ rbenv install X.Y.Z
+  ```
+- Linux
+
+  ```
+  $ sudo apt-get update
+  $ sudo apt-get rbenv ruby-build
+  $ rbenv install X.Y.Z
+  ```
+
+#### Installing PostgreSQL
+
+- Windows:
+
+  ```
+  C:> choco install postgresql
+  ```
+- Mac OS X
+
+  ```
+  $ brew update
+  $ brew install postgresql
+  ```
+- Linux
+  
+  ```
+  $ sudo apt-get update  
+  $ apt-get install postgresql postgresql-contrib
+  ```
+
+#### Installing Node.js (and npm)
 
-### To run the application inside a Vagrant virtual machine
+- Windows
 
-_Skip this step if you are not using Vagrant. The following assumes your Vagrantfile is configured to forward port 3000 to 3030, adjust the port number to suit your environment._
+  ```
+  C:> choco install nodejs
+  ```
+- Mac OS X
 
-If you are using a different port than 3000 to forward outside of your Vagrant environment you will need to go into `Procfile_Vagrant` and adjust the port number that applies to your needs.
+  ```
+  $ brew install node
+  ```
+- Linux
+  
+  *(see also [installing Node via package manager](https://nodejs.org/en/download/package-manager/).)*
+  ```
+  $ sudo apt-get install nodejs
+  ```
+- For other systems, see the [Node.js docs](https://github.com/joyent/node/wiki/Installing-Node.js-via-package-manager).
 
-* Copy the Vagrant Procfile example `cp Procfile_Vagrant.example Procfile_Vagrant`
-* Start the server with: `foreman s -f Procfile_Vagrant` (the `-f` explicitly states which Procfile to use and we don't want to use the original Procfile in our vagrant environments)
-* Sometimes you need to: `bundle exec foreman s -f Procfile_Vagrant`
-* Then you can access the local server at [localhost:3030](http://localhost:3030).
-* You can log in as a test user using the `assume` dropdown menu on the top right of the page without creating any new user for the app.
+With this, we're ready to download, build and run exercism.io locally (skip down to [The Code](#get-the-source-code))...
 
-_Again this is assuming you are forwarding port 3000 to 3030 in your Vagrantfile, adjust accordingly to your environment_
+### Running exercism.io in a Docker container via Vagrant
```

Comment:
At the start of this "tweak", I was aiming to re-organize the existing instructions (which didn't caveat using vagrant).

Yeah, the Docker readme is really nice... almost just want to point them to that and call it done.

This is really helpful, @tejasbubane!  I haven't used the Docker-based setup and have been kinda "wingin' it" with these here instructions.  I'll find a way to keep the main flow of instructions clear while handling these differences.